### ISSUE TITLE
allow pitch range per deck

### ISSUE THREAD

**Issue by ronso0** [mixxxdj/mixxx#12109]
Currently the selected rate range is applied to all decks, even though it can already be set per deck, for example in controller scripts.
So this is actually only about the Preferences:
* rate range selector for main decks 1-4
* rate range selector for samplers?

**Comment by Swiftb0y** [comment]
I think this is related to the more general issue where we have settings/COs per Deck but people also want a global CO/setting. In this case its the case that people want more fine-grained control per deck easily. In the case of selecting the QuickEffect, its the other way around... I think in order to solve this elegantly, we need some abstraction that unifies both cases.

**Comment by ronso0** [comment]
Could you explain that? I understand the situation, though what kind of abstraction are you referring to?
What other controls that come to mind?

**Comment by Swiftb0y** [comment]
In this case I just mean a general solution that doesn't have to be reimplemented for every single CO. Something like `GroupCO(COKey groupCOname, double nonUniformVal, QList<COKey> groupMembers)`, where `groupCOname` is the name of the overarching CO, `nonUniformVal` is the value of the CO when COs of the `groupMembers` don't agree with each other, and `groupMembers` is a collection of COs within. 
The behavior of the resulting CO is that its whatever value any member of `groupMembers` is when they all agree with each other, and `nonUniformVal` if they differ.

COs that come to mind would be most `[QuickEffectRack1_[ChannelI]]`, for compatibility with the global QuickEffects concept of most controllers. 

This feature applied to `[ChannelN], rateRange` would simplify the implementation of the current global rateRange implementation in the preferences. Though it doesn't directly solve the original issue.

**Comment by ronso0** [comment]
Ok, so like #11378. Got it, thx.

**Comment by Swiftb0y** [comment]
yes, exactly, just as a general implementation. I don't know when/whether I'll finish #11378.
